src/sleep/sleep.c

```diff
@@ -350,10 +350,18 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
 }
 
 static int execute_s2h(const SleepConfig *sleep_config) {
+        _cleanup_(freeze_thaw_user_slice) const char *method = "thaw";
+        const char *method_freeze = "freeze";
         int r, k;
 
         assert(sleep_config);
 
+        r = freeze_thaw_user_slice(&method_freeze);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to freeze unit user.slice, ignoring: %m");
+                method = NULL;
```

Comment:
it appears to me this is the wrong way round. instead of *disabling* the automatic thawing via `method = NULL` here in the error path I think we should only *enable* it in the success path, i.e. do:

```
r = freeze_thaw_user_slice(…);
if (r < 0)
        log_debug_errno(…);
else
        auto_thaw_method = "ThawUnit"; /* from now on we want automatic thawing */
```

Reply:
@poettering Yes automatic thawing is good but in case freeze was not done, don't we need to skip cleanup?

I will set it to NULL then where declaring the cleanup, makes sense right?

Reply:
what I am saying is that `auto_method_thaw` should default to NULL initially, and then only be set to `"ThawUnit"` once we the freezing succeeded.

Reply:
yes. done the same. :)